cpp/biconnected_components_module/algorithm/biconnected_components.hpp

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include <stack>
+
+#include <mg_graph.hpp>
+
+namespace bcc_utility {
+
+/// Simple struct that keeps the state of nodes in algorithms that rely on
+/// DFS traversal.
+struct NodeState {
+  std::vector<bool> visited;
+  std::vector<std::uint64_t> discovery, low_link, depth;
+  std::vector<std::uint64_t> parent;
+  uint64_t counter;
+
+  NodeState(std::uint64_t number_of_nodes) {
```

Comment:
Make it explicit. Good practice to use that keyword as much as possible so you don't create objects when you didn't want to.

Reply:
After reading about `explicit`, is there an example where we do not want to use it?

Reply:
I would correct myself a bit and say use the keyword ALWAYS for the single argument constructors to avoid accidental conversions.

For example, for the copy constructor it doesn't make much sense to have it explicit in any case.

For user defined constructors that take multiple arguments `explicit` will force you to define the name of function every time and you can't use the shorthand version without the class name.
https://godbolt.org/z/n6GchhfdY